### PR TITLE
replace spew methods with dump methods

### PR DESCRIPTION
**Pull request by haoruan**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Continue the cleanup https://github.com/kubernetes/kubernetes/pull/114800#issuecomment-1467157502

There're still some callings to `spew`, replace them with `dump.Pretty`,  so `dump` is the only place to use `spew`
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

/cc @thockin

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/dump/dump.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ var prettyPrintConfigForHash = &spew.ConfigState{
 	DisableCapacities:       true,
 }
 
-// Pretty wrap the spew.Sdump with Indent, and disabled methods like error() and String()
```
[the remark this answers is not in the document]
Comment (haoruan) [reply staging/src/k8s.io/apimachinery/pkg/util/dump/dump.go]: Oops, fixed

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**